docker/startup.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+CONFIG_FILE="/usr/share/nginx/html/config.json"
+
+# Check if config.json exists
+if [ -f "$CONFIG_FILE" ]; then
+  # Check if KONG_ADMIN_URL is set
+  if [ -n "$KONG_ADMIN_URL" ]; then
+    # Update the existing config.json with KONG_ADMIN_URL only
+    jq --arg kong_admin_url "$KONG_ADMIN_URL" \
+       '.kong_admin_url = $kong_admin_url' \
+       $CONFIG_FILE > tmp_file && mv tmp_file $CONFIG_FILE
+    echo "Updated config.json with KONG_ADMIN_URL"
+  fi
+
+  # Check if USERNAME, PASSWORD, and AUTH_TYPE are set
+  if [ -n "$USERNAME" ] && [ -n "$PASSWORD" ] && [ -n "$AUTH_TYPE" ]; then
+    # Update the existing config.json with USERNAME, PASSWORD, and AUTH_TYPE
+    jq --arg username "$USERNAME" \
+       --arg password "$PASSWORD" \
+       --arg auth_type "$AUTH_TYPE" \
+       '.username = $username | .password = $password | .auth_type = $auth_type' \
+       $CONFIG_FILE > tmp_file && mv tmp_file $CONFIG_FILE
+    echo "Updated config.json with USERNAME, PASSWORD, and AUTH_TYPE"
+  else
+    # Output an error if USERNAME, PASSWORD, or AUTH_TYPE is missing
+    echo "Error: USERNAME, PASSWORD, and AUTH_TYPE must be set"
+  fi
+else
+  # Create config.json if KONG_ADMIN_URL and all other variables are set
```

Comment:
To improve the security, only the user executing nginx should be able to read this config.json file. I'm not sure which user executes nginx in the alpine image but I think it's "nginx" (https://hub.docker.com/_/nginx).

Something like this should work if "nginx" is the user:
chmod 400 config.json
chown nginx:nginx config.json

Reply:
I dont think this will change much, since the application is running in the browser, it will need to be available through nginx anyways.. and this is much easier to access then having to get into the image and read a file...

Reply:
True... I was thinking "inside the box" and didn't think that anyways, the file is being served for anyone :P